src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -1488,69 +1488,116 @@ private static (Vector128<byte>, Vector128<byte>, Vector128<byte>) FormatGuidVec
         // IComparisonOperators
         //
 
-        /// <inheritdoc cref="IComparisonOperators{TSelf, TOther, TResult}.op_LessThan(TSelf, TOther)" />
-        public static bool operator <(Guid left, Guid right)
+        private static void GetMostSignificantDifference(in Guid left, in Guid right, out uint leftDifference, out uint rightDifference)
```

Comment:
This is a lot of additional code and complexity to the algorithm. Do you have any data showing that it is actually a hot path?

What about how this would perform as compared to something which just efficiently gets the value as a `UInt128` and uses those comparisons instead. For example:
```csharp
[MethodImpl(MethodImplOptions.AggressiveInlining)]
public static UInt128 AsUInt128(Guid guid)
{
    if (BitConverter.IsLittleEndian)
    {
        return new UInt128(
            ((ulong)(uint)guid._a << 32) | uint.RotateLeft(Unsafe.ReadUnaligned<uint>(ref Unsafe.As<short, byte>(ref Unsafe.AsRef(in guid._b))), 16),
            BinaryPrimitives.ReverseEndianness(Unsafe.ReadUnaligned<ulong>(ref Unsafe.AsRef(in guid._d)))
        );
    }
    else
    {
        return Unsafe.BitCast<Guid, UInt128>(guid);
    }
}

public static bool operator >(Guid left, Guid right)
{
    return AsUInt128(left) > AsUInt128(right);
}
```

Then, if you update the operators in `UInt128` to have 1 branch:
```csharp
public static bool operator >(UInt128 left, UInt128 right)
{
    return (left._upper != right._upper)
         ? (left._upper > right._upper)
         : (left._lower > right._lower);
}
```

You get the following highly compact and efficient codegen, all while simplifying the total code we have to maintain and reducing the number of branches:
```asm
    L0000: mov eax, [rcx]
    L0002: mov r8d, [rcx+4]
    L0006: mov rcx, [rcx+8]
    L000a: shl rax, 0x20
    L000e: rol r8d, 0x10
    L0012: or rax, r8
    L0015: bswap rcx
    L0018: mov r8d, [rdx]
    L001b: mov r10d, [rdx+4]
    L001f: mov rdx, [rdx+8]
    L0023: shl r8, 0x20
    L0027: rol r10d, 0x10
    L002b: or r8, r10
    L002e: bswap rdx
    L0031: cmp rax, r8
    L0034: jne short L0040
    L0036: xor eax, eax
    L0038: cmp rcx, rdx
    L003b: seta al
    L003e: jmp short L0049
    L0040: cmp rax, r8
    L0043: seta al
    L0046: movzx eax, al
    L0049: ret
```